lib/push.js

```diff
@@ -265,53 +195,43 @@ module.exports = function (log, db, config) {
     isValidPublicKey,
 
     /**
-     * Notify devices that there was an update to the account
+     * Notify devices that a new message is ready to be retrieved.
      *
      * @param {String} uid
-     * @param {Device[]} devices
+     * @param {Device} device
+     * @param {Device} url - url to retrieve the message
+     * @param {String} topic
      * @param {String} reason
-     * @param {Object} [options]
-     *   @param {String[]} [options.includedDeviceIds]
-     *   @param {String[]} [options.excludedDeviceIds]
-     *   @param {Object} [options.data]
-     *   @param {String} [options.TTL] (in seconds)
      * @promise
      */
-    notifyUpdate (uid, devices, reason, options = {}) {
-      if (options.includedDeviceIds) {
-        const include = new Set(options.includedDeviceIds)
-        devices = devices.filter(device => include.has(device.id))
-
-        if (devices.length === 0) {
-          return P.reject('devices empty')
-        }
-      } else if (options.excludedDeviceIds) {
-        const exclude = new Set(options.excludedDeviceIds)
-        devices = devices.filter(device => ! exclude.has(device.id))
+    notifyMessageReceived(uid, device, url, topic, reason) {
+      const options = {
+        data: {
+          url,
+          topic
+        },
+        TTL: TTL_MESSAGE_RECEIVED
```

Comment:
I don't understand why this push notification appears to be structured differently to the others.  In other cases the body is structured like this:

```
data: {
     version: PUSH_PAYLOAD_SCHEMA_VERSION,
     command: PUSH_COMMANDS.PASSWORD_RESET
 },
 TTL: TTL_PASSWORD_RESET
```

But the "message received" one doesn't seem to have "version" or "command" fields.  Am I missing something that means this has to be different?

Reply:
I'm thinking of clients trying to handle their push message stream using `switch (message["commend"]) { ... }` but not having a "command" field to switch on in this case.